api/api/controllers/agents_controller.py

```diff
@@ -424,57 +442,79 @@ async def put_upgrade_agent(request, agent_id, pretty=False, wait_for_complete=F
                           f_kwargs=remove_nones_to_dict(f_kwargs),
                           request_type='distributed_master',
                           is_async=False,
-                          wait_for_complete=True,  # Force wait_for_complete until timeout problems are resolved
+                          wait_for_complete=wait_for_complete,
                           logger=logger,
+                          broadcasting=agents_list == '*',
```

Comment:
Why are we broadcasting this call if we receive a list of agents?

Reply:
broadcasting removed